docs/dev/dod.md

```diff
@@ -0,0 +1,14 @@
+# Definition of Done
+These criteria must be met before a feature or artifact can be considered "done".
+
+## Code
+- Code coverage: >75%
+- Codacy grade A
+- Travis builds the project and runs all tests without error
```

Comment:
Maybe, you should distinguish between module tests and integration tests. Normal software projects, don't run in timeout issues on travis.

Further, matrix builds can help: https://docs.travis-ci.com/user/speeding-up-the-build/. One can do tests in parallel.